lib/ot-date.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+module.exports = function (moment) {
+    /**
+     * Open Token Date Class to provide simple date functionality
+     * using Moment.js.
+     *
+     * @param {Moment} m
+     */
+    class OtDate {
+        constructor(m) {
+            if (! (m instanceof moment)) {
+                throw new Error("Use helper functions instead");
+            }
+
+            this.moment = m.utc();
+        }
+
+
+        /**
+         * Checks if the set time is before the time passed in.
+         *
+         * @param {Date} otherDate
+         * @return {boolean} whether the date is before
+         */
+        isBefore(otherDate) {
+            return this.moment.isBefore(otherDate);
+        }
+
+
+        /**
+         * Adds time to the set time.
+         * You can send in multiple parameters to move the date up.
+         *
+         *    {"days": 1, "hours": 5, "seconds": 3}
+         *
+         * @param {Object} spec
+         * @return {Moment}
```

Comment:
I don't ever want to expose the `moment` object outside.  This should not return or `return this`